src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FilterDelegationBase.cs

```diff
@@ -114,6 +122,16 @@ protected bool IsValidDelegatableFilterPredicateNode(TexlNode dsNode, TexlBindin
 
                     case NodeKind.DottedName:
                         {
+                            // Only boolean option set, boolean fields and views are allowed to delegate
+                            var nodeDType = binding.GetType(dsNode);
+                            if (!((nodeDType.IsOptionSet && nodeDType.OptionSetInfo != null && nodeDType.OptionSetInfo.IsBooleanValued) 
+                                || (nodeDType == DType.Boolean && dsNode.Kind != NodeKind.BoolLit)
```

Comment:
replace the first two lines of this condition with `DType.Boolean.Accepts(nodeDType)`

Same goes for the condition on 109.

Reply:
and the one in opdelstrategy

Reply:
Whoops, I was thinking of CoercesTo, which is more than you're looking for.